### PR TITLE
Add `last_day` Function To OpenSearch SQL Plugin

### PR DESCRIPTION
**Pull request by GabeFernandez310**
### Description
Adds the last_day function to the OpenSearch SQL plugin according to [MySQL documentation](https://dev.mysql.com/doc/refman/8.0/en/date-and-time-functions.html#function_last-day).

 
### Issues Resolved
https://github.com/opensearch-project/sql/issues/722
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: integ-test/src/test/java/org/opensearch/sql/sql/DateTimeFunctionIT.java [hunk 1]
```diff
@@ -442,6 +442,16 @@ public void testHourFunctionAliasesReturnTheSameResults() throws IOException {
     result1.getJSONArray("datarows").similar(result2.getJSONArray("datarows"));
   }
 
+  @Test
+  public void testDayLastDay() throws IOException {
+    JSONObject result = executeQuery(
+        String.format("SELECT last_day(CAST(date0 AS date)) FROM %s LIMIT 3", TEST_INDEX_CALCS));
```
Comment (acarbonetto) [inline integ-test/src/test/java/org/opensearch/sql/sql/DateTimeFunctionIT.java:448]: do you need to cast?

**Review by acarbonetto — APPROVED** [review]
LGMT

**Review by forestmvey — APPROVED** [review]

**Review by Yury-Fridlyand — APPROVED** [review]

File: integ-test/src/test/java/org/opensearch/sql/sql/DateTimeFunctionIT.java [hunk 2]
```diff
@@ -513,6 +513,16 @@ public void testHourFunctionAliasesReturnTheSameResults() throws IOException {
     result1.getJSONArray("datarows").similar(result2.getJSONArray("datarows"));
   }
 
+  @Test
+  public void testLastDay() throws IOException {
+    JSONObject result = executeQuery(
+        String.format("SELECT last_day(date0) FROM %s LIMIT 3", TEST_INDEX_CALCS));
```
Comment (GumpacG) [inline integ-test/src/test/java/org/opensearch/sql/sql/DateTimeFunctionIT.java:519]: Could add `time0` and `datetime0` to test time and timestamp.

**Review by GumpacG — APPROVED** [review]

File: integ-test/src/test/java/org/opensearch/sql/sql/DateTimeFunctionIT.java (hunk 2 above)
Comment (Yury-Fridlyand) [reply integ-test/src/test/java/org/opensearch/sql/sql/DateTimeFunctionIT.java:519]: +1
with cast to ensure that different types are supplied to the function

**Review by Yury-Fridlyand — APPROVED** [review]

**Review by matthewryanwells — APPROVED** [review]

File: integ-test/src/test/java/org/opensearch/sql/sql/DateTimeFunctionIT.java (hunk 2 above)
Comment (GabeFernandez310) [reply integ-test/src/test/java/org/opensearch/sql/sql/DateTimeFunctionIT.java]: Added here using casting to ensure that the types passed in are correct https://github.com/Bit-Quill/opensearch-project-sql/pull/218/commits/22be4c7d0ff0a56469bb6f90383b7a35daf23e16

**Pull request merged**